src/mscorlib/src/System/CurrentTimeZone.cs

```diff
@@ -107,119 +89,124 @@ internal class CurrentSystemTimeZone : TimeZone {
+            
+            if (isDst)
+            {
                 offset += daylightTime.Delta;
                 
                 // See if the resulting local time becomes ambiguous. This must be captured here or the
                 // DateTime will not be able to round-trip back to UTC accurately.
-                if (time >= ambiguousStart && time < ambiguousEnd ) {
+                if (time >= ambiguousStart && time < ambiguousEnd )
+                {
                     isAmbiguousLocalDst = true;
                 }
             }
             return offset.Ticks;
         }
         
-        public override DateTime ToLocalTime(DateTime time) {
-            if (time.Kind == DateTimeKind.Local) {
+        public override DateTime ToLocalTime(DateTime time)
+        {
+            if (time.Kind == DateTimeKind.Local)
+            {
                 return time;
             }
             Boolean isAmbiguousLocalDst = false;
             Int64 offset = GetUtcOffsetFromUniversalTime(time, ref isAmbiguousLocalDst);
             long tick = time.Ticks + offset;
-            if (tick>DateTime.MaxTicks) {
+            if (tick > DateTime.MaxTicks)
+            {
                 return new DateTime(DateTime.MaxTicks, DateTimeKind.Local);
             }
-            if (tick<DateTime.MinTicks) {
+            if (tick < DateTime.MinTicks)
+            {
                 return new DateTime(DateTime.MinTicks, DateTimeKind.Local);
             }
-            return new DateTime(tick, DateTimeKind.Local, isAmbiguousLocalDst);            
+            return new DateTime(tick, DateTimeKind.Local, isAmbiguousLocalDst);
         }
 
         // Private object for locking instead of locking on a public type for SQL reliability work.
         private static Object s_InternalSyncObject;
-        private static Object InternalSyncObject {
-            get {
-                if (s_InternalSyncObject == null) {
+        private static Object InternalSyncObject
+        {
+            get
+            {
+                if (s_InternalSyncObject == null)
+                {
                     Object o = new Object();
                     Interlocked.CompareExchange<Object>(ref s_InternalSyncObject, o, null);
                 }
                 return s_InternalSyncObject;
             }
         }
 
-
-        [System.Security.SecuritySafeCritical]  // auto-generated
-        public override DaylightTime GetDaylightChanges(int year) {
-            if (year < 1 || year > 9999) {
+        public override DaylightTime GetDaylightChanges(int year)
+        {
+            if (year < 1 || year > 9999)
+            {
                 throw new ArgumentOutOfRangeException(nameof(year), Environment.GetResourceString("ArgumentOutOfRange_Range", 1, 9999));
             }
             Contract.EndContractBlock();
 
-            Object objYear = (Object)year;
-
-            if (!m_CachedDaylightChanges.Contains(objYear)) {
-                lock (InternalSyncObject) {
```

Comment:
I didn't remove the lock, I moved it down so the whole block we are creating a new item doesn't have inside the lock block. note that the code already reading from the hash table before locking too. so I didn't really change the logic

---
In reply to: [87496264](https://github.com/dotnet/coreclr/pull/8074#discussion_r87496264) [](ancestors = 87496264,87495577,87494826)